Tests/ML/test_metrics.py

```diff
@@ -171,12 +172,12 @@ def test_classification_metrics() -> None:
     for output, label in zip(outputs, labels):
         for metric in metrics:
             metric.update(output, label)
-    accuracy_05, accuracy_opt, threshold, fpr, fnr, roc_auc, pr_auc, cross_entropy = [metric.compute() for metric in
+    accuracy_05, accuracy_opt, threshold, fpr, fnr, roc_auc, pr_auc, cross_entropy_with_logits = [metric.compute() for metric in
                                                                                       metrics]
     all_labels = torch.cat(labels).numpy()
     all_outputs = torch.cat(outputs).numpy()
     expected_accuracy_at_05 = np.mean((all_outputs > 0.5) == all_labels)
-    expected_binary_cross_entropy = log_loss(y_true=all_labels, y_pred=all_outputs)
+    expected_binary_cross_entropy_with_logits = log_loss(y_true=all_labels, y_pred=expit(all_outputs))
```

Comment:
The test logic is somewhat twisted now. You care calling `metric.update` with posteriors even on the one metric that needs logits. In line 180, you are then exponentiating a posterior to still make the numbers match.
It would be cleaner if you added the same check (compute_from_logits == True) to line 174.